### PR TITLE
refactor: changes to make assetTicker the pri key + remove assetId

### PR DESCRIPTION
**Pull request by julianooii**
Some minor changes to the SQL + java files to remove the assetId pri key and make assetTicker the pri key

### CODE REVIEW

File: src/main/java/com/backend/controller/PortfolioController.java [hunk 1]
```diff
@@ -98,27 +98,33 @@ public CreatePortfolioAssetResponse addPortfolioAsset(@RequestBody CreatePortfol
...
-		if (request.getAssetId() == null) {
+		if (request.getAssetTicker() == null) {
 			throw new BadRequestException(Constants.MESSAGE_INVALIDASSETID);
 		}
 
+		// Portfolio portfolio = restTemplate.exchange("http://localhost:8080/portfolio/"+request.getPortfolioId(), HttpMethod.GET, null, Portfolio.class).getBody();
+
+		
+		// logger.info("Adding " + request.getAssetId() + " to portfolio " + portfolio.getPid());
+		logger.info("New Portfolio Asset ID: " + request.getAssetTicker());
```
Comment (lionelgwk) [inline src/main/java/com/backend/controller/PortfolioController.java:109]: Should this be "New Portfolio Asset Ticker:" ?

File: src/main/java/com/backend/controller/PortfolioController.java [hunk 2]
```diff
@@ -98,27 +98,33 @@ public CreatePortfolioAssetResponse addPortfolioAsset(@RequestBody CreatePortfol
 		if (request.getPortfolioId() == null) {
 			throw new BadRequestException(Constants.MESSAGE_INVALIDPORTFOLIOID);
 		}
-		if (request.getAssetId() == null) {
+		if (request.getAssetTicker() == null) {
 			throw new BadRequestException(Constants.MESSAGE_INVALIDASSETID);
```
Comment (lionelgwk) [inline src/main/java/com/backend/controller/PortfolioController.java:102]: Should we change the constant error message as well for `MESSAGE_INVALIDASSETID`?

File: src/main/java/com/backend/controller/PortfolioController.java [hunk 3]
```diff
@@ -98,27 +98,33 @@ public CreatePortfolioAssetResponse addPortfolioAsset(@RequestBody CreatePortfol
...
 						request.getAveragePrice(),
 						request.getQuantity()));
 
 		java.util.Date time = new java.util.Date(portfolioAsset.getDateCreated() * 1000);
 		SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
-		CreatePortfolioAssetResponse response = new CreatePortfolioAssetResponse();
+		CreatePortfolioAssetResponse response = new CreatePortfolioAssetResponse(); 
 
 		logger.info("Created on: " + time);
-		logger.info("New Portfolio Asset ID: " + request.getAssetId());
```
Comment (lionelgwk) [inline src/main/java/com/backend/controller/PortfolioController.java:117]: After this deletion, should this be replaced with `logger.info("New Portfolio Asset Ticker: " + request.getAssetTicker());` ?

File: src/main/java/com/backend/service/concretions/AssetService.java [hunk 4]
```diff
@@ -17,13 +16,9 @@ public AssetService(AssetRepository repository) {
...
     @Override
-    public Asset findByAssetId(long assetId) {
-        Asset asset = repository.findByAssetId(assetId);
-        if (asset == null) {
-            throw new AssetNotFoundException(assetId);
-        } else {
-            return asset;
-        }
+    public Asset findByAssetTicker(String assetTicker){
+        Asset asset = repository.findByAssetTicker(assetTicker);
```
Comment (lionelgwk) [inline src/main/java/com/backend/service/concretions/AssetService.java:20]: Should we have error handling here as well?

File: src/main/java/com/backend/controller/PortfolioController.java (hunk 3 above)
Comment (julianooii) [reply src/main/java/com/backend/controller/PortfolioController.java:167]: oki done!

**Pull request merged**